8339791: Refactor MiscUndecorated/ActiveAWTWindowTest.java

srmandal opened the pull request:
The java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java test uses object monitors and wait/notify to synchronise actions in the test.

Using CountDownLatch could make the test simpler, shorter, clearer.
Tested the code on a windows-x64, macos-x64 and lnux-x64 machines and the test is working as expected.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339791](https://bugs.openjdk.org/browse/JDK-8339791): Refactor MiscUndecorated/ActiveAWTWindowTest.java (**Bug** - P4)


### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/26471/head:pull/26471` \
`$ git checkout pull/26471`

Update a local copy of the PR: \
`$ git checkout pull/26471` \
`$ git pull https://git.openjdk.org/jdk.git pull/26471/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 26471`

View PR using the GUI difftool: \
`$ git pr show -t 26471`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/26471.diff">https://git.openjdk.org/jdk/pull/26471.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/26471#issuecomment-3116468520)
</details>

kumarabhi006 commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -93,13 +97,7 @@ private void initializeGUI() {
         frame.addWindowFocusListener(new WindowFocusListener() {
             public void windowGainedFocus(WindowEvent event) {
```
add @override for all overridden methods.

kumarabhi006 reviewed (commented):
Few more observations :
1. Add a space wherever application. At L76... `try {`
2. Don't think it is required to print stack trace, you can throw it in RTE.
3. At L165, robot.waitForIdle(5 * delay);
4. Access UI components on EDT. e.g L166 etc.
5. Can remove `//captureScreenAndSave();` and `System.err.println("Test failed!");` lines

srmandal commented:
All Comments have been incorporated. 
1. Replaced ExtendedRobot by Robot
2. focusGainLatch usage fixed with an appropriate countDown
3. Replaced Delay*N with actual values.
4. Removed printStackTrace.
5. UI components access restricted to EDT fixed, both initiatiliseGUI, doTest are now static.
6. @Override added
7. Also the frame.toFront is now removed, as there is already a mouse click on the frame in the code, which is more reliable and has same effect - for the test scenario.

prsadhuk commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -160,75 +146,68 @@ public void actionPerformed(ActionEvent e) {
+            try {
+                windowActivatedLatch.await(1500, TimeUnit.MILLISECONDS);
+            } catch (Exception e) {
+                throw new RuntimeException("Unexpected Exception: "
```
same

prsadhuk commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -160,75 +146,68 @@ public void actionPerformed(ActionEvent e) {
+            try {
+                windowDeactivatedLatch.await(1500, TimeUnit.MILLISECONDS);
+            } catch (Exception e) {
+                throw new RuntimeException("Unexpected Exception: "
```
same

srmandal commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -160,75 +146,68 @@ public void actionPerformed(ActionEvent e) {
+            if (!windowFocusGainedLatch.await(1500, TimeUnit.MILLISECONDS)) {
+                passed = false;
+            }
+        } catch (InterruptedException e) {
```
Changed as suggested.

lawrence-andrew commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -149,9 +129,11 @@ public void actionPerformed(ActionEvent e) {
         button2 = new Button("Click me");
         textField2 = new TextField("TextField");
         button2.addActionListener(new ActionListener() {
+            @Override
```
Can this be converted to lambda

srmandal commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -24,116 +24,96 @@
         textField = new TextField("TextField");
         button = new Button("Click me");
         button.addActionListener(new ActionListener() {
+            @Override
```
Done

srmandal commented on test/jdk/java/awt/Frame/MiscUndecorated/ActiveAWTWindowTest.java:
```diff
@@ -160,75 +142,47 @@ public void actionPerformed(ActionEvent e) {
-        robot.waitForIdle(5*delay);
-        robot.mouseMove(button.getLocationOnScreen().x + button.getSize().width / 2,
-                        button.getLocationOnScreen().y + button.getSize().height / 2);
+        if (!windowFocusGainedLatch.await(1500, TimeUnit.MILLISECONDS)) {
```
Yes, that was the code initially. The previous review suggested removal of the code as being redundant in the presence of the main method's throws Exception. There are pros and con to argue on both ways. Please let me know if you strong reason to not ok to not have in the current form.

srmandal commented:
/integrate

prsadhuk commented:
/sponsor
